src/erlcloud_inspector.erl

```diff
@@ -0,0 +1,1155 @@
+          (RunArn :: string(),
+           Config :: aws_config()) ->
+          return_val().
+describe_run(RunArn, Config) ->
+    Json = [{<<"runArn">>, to_binary(RunArn)}],
+    inspector_request(Config, "InspectorService.DescribeRun", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% DetachAssessmentAndRulesPackage
+%%
+%% http://docs.aws.amazon.com/inspector/latest/APIReference/API_DetachAssessmentAndRulesPackage.html
+%%%------------------------------------------------------------------------------
+
+-spec detach_assessment_and_rules_package/2 ::
+          (AssessmentArn :: string(),
+           RulesPackageArn :: string()) ->
+          return_val().
+detach_assessment_and_rules_package(AssessmentArn, RulesPackageArn) ->
+    detach_assessment_and_rules_package(AssessmentArn, RulesPackageArn, default_config()).
+
+
+-spec detach_assessment_and_rules_package/3 ::
+          (AssessmentArn :: string(),
+           RulesPackageArn :: string(),
+           Config :: aws_config()) ->
+          return_val().
+detach_assessment_and_rules_package(AssessmentArn, RulesPackageArn, Config) ->
+    Json = [{<<"assessmentArn">>, to_binary(AssessmentArn)},
+            {<<"rulesPackageArn">>, to_binary(RulesPackageArn)}],
+    inspector_request(Config, "InspectorService.DetachAssessmentAndRulesPackage", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetAssessmentTelemetry
+%%
+%% http://docs.aws.amazon.com/inspector/latest/APIReference/API_GetAssessmentTelemetry.html
+%%%------------------------------------------------------------------------------
+
+-spec get_assessment_telemetry/1 ::
+          (AssessmentArn :: string()) ->
+          return_val().
+get_assessment_telemetry(AssessmentArn) ->
+    get_assessment_telemetry(AssessmentArn, default_config()).
+
+
+-spec get_assessment_telemetry/2 ::
+          (AssessmentArn :: string(),
+           Config :: aws_config()) ->
+          return_val().
+get_assessment_telemetry(AssessmentArn, Config) ->
+    Json = [{<<"assessmentArn">>, to_binary(AssessmentArn)}],
+    inspector_request(Config, "InspectorService.GetAssessmentTelemetry", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% ListApplications
+%%
+%% http://docs.aws.amazon.com/inspector/latest/APIReference/API_ListApplications.html
+%%%------------------------------------------------------------------------------
+
+-spec list_applications/0 ::
+          () ->
+          return_val().
+list_applications() ->
+    list_applications([]).
+
+
+-spec list_applications/1 ::
+          (Options :: inspector_opts()) ->
+          return_val().
+list_applications(Options) ->
+    list_applications(Options, default_config()).
+
+
+-spec list_applications/2 ::
+          (Options :: inspector_opts(),
+           Config :: aws_config()) ->
+          return_val().
+list_applications(Options, Config) ->
```

Comment:
```{next_token, NextToken :: binary()}``` can be passed as an option.

I've implemented this so that mandatory arguments are mandatory request parameters for the API and anything that is optional goes into the options proplist. This is a much cleaner way IMO, to approach the client, and I think a step above the current aws sdks, which accept everything as an array or list of arguments, regardless if the parameter is required or optional.